test/Microsoft.Extensions.Internal.Test/TypeNameHelperTest.cs

```diff
@@ -9,110 +9,124 @@ namespace Microsoft.Extensions.Internal
-            Assert.Equal("Microsoft.Extensions.Internal.TypeNameHelperTest+Outer<int>+F<int, string>",
-                TypeNameHelper.GetTypeDisplayName(typeof(Outer<int>.F<int, string>)));
-            Assert.Equal("Microsoft.Extensions.Internal.TypeNameHelperTest+Outer<int>+F<int, Microsoft.Extensions.Internal.TypeNameHelperTest+Outer<int>+E<string>>",
-                TypeNameHelper.GetTypeDisplayName(typeof(Outer<int>.F<int, Outer<int>.E<string>>)));
-            Assert.Equal("Microsoft.Extensions.Internal.TypeNameHelperTest+Outer<int>+E<Microsoft.Extensions.Internal.TypeNameHelperTest+Outer<int>+E<string>>",
-                TypeNameHelper.GetTypeDisplayName(typeof(Outer<int>.E<Outer<int>.E<string>>)));
-            Assert.Equal("Microsoft.Extensions.Internal.TypeNameHelperTest+OuterGeneric<int>+InnerNonGeneric+InnerGeneric<int, string>+InnerGenericLeafNode<bool>",
-                TypeNameHelper.GetTypeDisplayName(typeof(OuterGeneric<int>.InnerNonGeneric.InnerGeneric<int, string>.InnerGenericLeafNode<bool>)));
-            Assert.Equal("Microsoft.Extensions.Internal.TypeNameHelperTest+Level1<int>+Level2<bool>+Level3<int>",
-                TypeNameHelper.GetTypeDisplayName(typeof(Level1<int>.Level2<bool>.Level3<int>)));
+            yield return new object[] { "Microsoft.Extensions.Internal.TypeNameHelperTest+Outer<int>+D", typeof(Outer<int>.D) };
+            yield return new object[] { "Microsoft.Extensions.Internal.TypeNameHelperTest+Outer<int>+E<int>", typeof(Outer<int>.E<int>) };
+            yield return new object[] { "Microsoft.Extensions.Internal.TypeNameHelperTest+Outer<int>+F<int, string>", typeof(Outer<int>.F<int, string>) };
+
+            yield return new object[]
+            {
+                "Microsoft.Extensions.Internal.TypeNameHelperTest+Outer<int>+F<int, Microsoft.Extensions.Internal.TypeNameHelperTest+Outer<int>+E<string>>",
+                typeof(Outer<int>.F<int, Outer<int>.E<string>>)
+            };
+
+            yield return new object[]
+            {
+                "Microsoft.Extensions.Internal.TypeNameHelperTest+Outer<int>+E<Microsoft.Extensions.Internal.TypeNameHelperTest+Outer<int>+E<string>>",
+                typeof(Outer<int>.E<Outer<int>.E<string>>)
+            };
+
+            yield return new object[]
+            {
+                "Microsoft.Extensions.Internal.TypeNameHelperTest+OuterGeneric<int>+InnerNonGeneric+InnerGeneric<int, string>+InnerGenericLeafNode<bool>",
+                typeof(OuterGeneric<int>.InnerNonGeneric.InnerGeneric<int, string>.InnerGenericLeafNode<bool>)
+            };
+
+            yield return new object[]
+            {
+                "Microsoft.Extensions.Internal.TypeNameHelperTest+Level1<int>+Level2<bool>+Level3<int>",
+                typeof(Level1<int>.Level2<bool>.Level3<int>)
+            };
         }
 
-        [Fact]
-        public void Can_pretty_print_CLR_name()
+        [Theory]
+        [MemberData(nameof(GetFullTypeNamesTestData))]
```

Comment:
Either put `InlineData` or just keep the tests as is. Not a fan of `MemberData` for readability purposes.

Reply:
If arguments are too big for `InlineData` we use `MemberData` in many places, it's much better then putting all tests cases in one `[Fact]`

Reply:
But If we make it a single test then execution will stop after first error and we won't get info about all of them, also inside `GetFullTypeNamesTestData` every test case clearly distinguishable and readable and it`s near with test itself. I'll try to put it in the `InlineData`, but some lines could be quite bulky.

Reply:
Leave MemberData as is :)

Reply:
Thanks, I've tried option with `InlineData` and it actually not that bad, if [this](https://gist.github.com/AndreyTretyak/57623e99c4596c6f16ae8109edaceee7) is acceptable I can commit it.